File: src/main/java/it/tylframework/data/mongo/basics/Country.java
package it.tylframework.data.mongo.basics;

import it.tylframework.data.mongo.common.Footprint;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NonNull;
import lombok.RequiredArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.TypeAlias;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Date;

/**
 * Created by mp on 20/11/14.
 */
@Document(collection="bas_country")
@TypeAlias("Country")
@Data
@RequiredArgsConstructor
public class Country{
    @NonNull String twoCharCode;
    @NonNull String officialName;
    @NonNull Integer numericCode;

    @CreatedDate
    private Date createdDate;
}

File: src/main/java/it/tylframework/data/mongo/basics/Country.kt
package it.tylframework.data.mongo.basics

import it.tylframework.data.mongo.common.Footprint
import it.tylframework.data.mongo.common.Signature
import java.util.Date
import org.springframework.data.annotation.TypeAlias

/**
 * Created by marco on 25/08/14.
 */
[TypeAlias("bas_Country")]
data class Country(
        var twoCharCode: String,
        var officialName:String,
        var numericCode:Int)
        :Footprint(Signature("marco"), Date() ,Signature("marco"),  Date()) {
}